spec/models/post_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  it 'is valid with valid attributes' do
+    user = User.create(name: 'John Doe', posts_counter: 0)
+    post = user.posts.build(title: 'Valid Post', comments_counter: 0, likes_counter: 0)
+    expect(post).to be_valid
+  end
+
+  it 'is not valid without a title' do
+    user = User.create(name: 'Alice', posts_counter: 0)
+    post = user.posts.build(comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'is not valid with a long title' do
+    user = User.create(name: 'Bob', posts_counter: 0)
+    post = user.posts.build(title: 'A' * 251, comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'is not valid with a negative comments_counter' do
+    user = User.create(name: 'Eve', posts_counter: 0)
+    post = user.posts.build(title: 'Negative Comments', comments_counter: -1, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'increments the user posts_counter after creation' do
+    user = User.create(name: 'Charlie', posts_counter: 0)
+    user.posts.create(title: 'New Post', comments_counter: 0, likes_counter: 0)
+    user.reload
+    expect(user.posts_counter).to eq(1)
+  end
+end
```

Comment:
- You have done well so far, but according to the project requirements, Any custom methods in the models should have a corresponding unit test. Please test for the `recent_comments` method in your post model 👍 👍